pkg/registry/apis/secret/decrypt/authorizer.go

```diff
@@ -69,16 +80,27 @@ func (a *decryptAuthorizer) Authorize(ctx context.Context, ns xkube.Namespace, s
 		return serviceIdentity, false
 	}
 
-	// Finally check whether the service identity is allowed to decrypt this secure value.
-	allowed := false
+	// Check whether the service identity is allowed to decrypt this secure value.
 	for _, decrypter := range secureValueDecrypters {
 		if decrypter == serviceIdentity {
-			allowed = true
-			break
+			return serviceIdentity, true
+		}
+	}
+
+	// finally check if the owner matches any hardcoded service identities
+	if owners != nil && a.extra != nil {
+		for _, extra := range a.extra {
+			if extra.Identity == serviceIdentity {
+				for _, owner := range owners {
```

Comment:
nit: slices.IndexFunc might be nice for at least one, maybe both of these for loops.